staging_vespalib/src/vespa/vespalib/metrics/stable_store.h

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include <assert.h>
+
+namespace vespalib {
+
+template <typename T>
+class StableStore
+{
+    using MyClass = StableStore<T>;
+    template<typename U>
+    friend void swap(StableStore<U> &a, StableStore<U> &b);
+public:
+    typedef std::unique_ptr<MyClass> UP;
+
+    StableStore();
+    ~StableStore() {}
+
+    void add(T t) {
+        size_t sz = _mine.size();
+        if (sz == _mine.capacity()) {
+            UP next(new MyClass(_size, std::move(_more), std::move(_mine)));;
```

Comment:
Consider `make_unique` to avoid naked new

Reply:
make_unique can't use the private constructor.